src/components/NewsletterForm/NewsletterForm.tsx

```diff
@@ -1,9 +1,46 @@
+'use client';
+
+import { useRef } from 'react';
+import { Button, FormControl, Input, Stack } from '@mui/material';
+
+import { useData } from '../../hooks/useData';
+
+import styles from './NewsletterForm.module.scss';
+
 export function NewsletterForm() {
+  const ref = useRef<HTMLFormElement>(null);
+  const { loading, message, setMail, Subscribe, success } = useData();
+
+  const Submit = (e: React.FormEvent<HTMLFormElement>) => {
```

Comment:
z dużej piszemy komponenty,
przyda się jeszcze jakaś podstawowa walidacja na ten email

Reply:
Dodałam w `inputProps` pattern dla emaila, to jest ok?